packages/syft/src/syft/service/queue/queue.py

```diff
@@ -78,31 +224,41 @@ def handle_message(message: bytes):
             document_store_config=worker_settings.document_store_config,
             action_store_config=worker_settings.action_store_config,
             blob_storage_config=worker_settings.blob_store_config,
+            queue_config=queue_config,
             is_subprocess=True,
+            migrate=False,
         )
+        # otherwise it reads it from env, resulting in the wrong credentials
+        worker.id = worker_settings.id
+        worker.signing_key = worker_settings.signing_key
 
-        item = QueueItem(
-            node_uid=worker.id,
-            id=task_uid,
-            status=Status.PROCESSING,
-        )
-        worker.queue_stash.set_result(api_call.credentials, item)
-        status = Status.COMPLETED
-
-        try:
-            result = worker.handle_api_call(api_call)
-            if isinstance(result, SyftError):
-                status = Status.ERRORED
-        except Exception as e:  # nosec
-            status = Status.ERRORED
-            result = SyftError(message=f"Failed with exception: {e}")
+        credentials = queue_item.syft_client_verify_key
 
-        item = QueueItem(
-            node_uid=worker.id,
-            id=task_uid,
-            result=result,
-            resolved=True,
-            status=status,
+        job_item = worker.job_stash.get_by_uid(credentials, queue_item.job_id).ok()
+
+        queue_item.status = Status.PROCESSING
+        queue_item.node_uid = worker.id
+
+        job_item.status = JobStatus.PROCESSING
+        job_item.node_uid = worker.id
+
+        worker.queue_stash.set_result(credentials, queue_item)
+        worker.job_stash.set_result(credentials, job_item)
```

Comment:
Instead of duplicating this we can instead just forward the message bytes to the process function `handle_message_multiprocessing` where we can run this routine followed by handling and setting results.

```
@staticmethod
def handle_message(message: bytes):
    p = multiprocessing.Process(
            target=handle_message_multiprocessing,
            args=(message),
        )
    p.start()
    p.join()
```

Reply:
Aditionally, we can have fixed size process pool of size "n", so that we can process "n" messages without blocking the worker